crates/repository/src/lib.rs

```diff
@@ -336,9 +338,22 @@ impl Repository {
                     .await
                     .wrap_err("failed to download SDK")?;
             }
+            if sdk.exists() {
+                println!("Removing incomplete SDK directory...");
+                remove_dir_all(&sdk)
+                    .await
+                    .wrap_err("failed to remove old SDK directory")?;
+            }
```

Comment:
Can we move this block outside the `if need_to_install` and guard it with `incomplete_marker.exists()`? If we then remove the directory first, the "old" check for `needs_to_install` by checking whether there is an SDK suffices, as we removed it...

Reply:
Good point, this seems cleaner.